share/functions/fish_config.fish

```diff
@@ -268,12 +303,23 @@ function fish_config --description "Launch fish's web based configuration"
                         end
 
                         if not set -q file[1]
-                            echo "No such theme: $argv[1]" >&2
-                            echo "Searched directories: $dirs" >&2
-                            return 1
+                            if status list-files tools/web_config/themes/$argv[1].theme >/dev/null
+                                set file tools/web_config/themes/$argv[1].theme
+                            else
+                                echo "No such theme: $argv[1]" >&2
+                                echo "Searched directories: $dirs" >&2
+                                return 1
+                            end
                         end
 
-                        while read -lat toks
+                        set -l content
+                        if string match -qr '^tools/' -- $file
+                            set content (status get-file $file)
+                        else
+                            read -az content < $file
+                        end
```

Comment:
something here must be broken. Visual mode selections are no longer highlighted. I'm  using

	target/debug/fish -C 'set -g fish_key_bindings fish_vi_key_bindings'

Confirmed that "set -S fish_color_selection" still shows "-r"

Reply:
How are you setting the theme?

When I do that I get my regular selection color and it works.

Reply:
I'm also not sure how that should be caused by this PR - it does not touch any of the highlighting, so if the variable is set I don't know how it should affect it.

Reply:
Ah, okay: It needs to be `read -z`, without the `-a`.

Note that that means you would have an *empty* fish_color_selection, not "-r". So presumably you have a universal set to "-r", and are doing "fish_config theme choose None", and that sets a global to empty.